tests/foreman/destructive/test_capsule_loadbalancer.py

```diff
@@ -113,96 +114,78 @@ def setup_capsules(
+    result = module_haproxy.execute('yum install haproxy policycoreutils-python-utils -y')
     assert result.status == 0
-    haproxy.execute('rm -f /etc/haproxy/haproxy.cfg')
-    haproxy.session.sftp_write(
+    module_haproxy.execute('rm -f /etc/haproxy/haproxy.cfg')
+    module_haproxy.session.sftp_write(
         source=DataFile.DATA_DIR.joinpath('haproxy.cfg'),
         destination='/etc/haproxy/haproxy.cfg',
     )
-    haproxy.execute(
-        f'sed -i -e s/CAPSULE_1/{setup_capsules["capsule_1"].hostname}/g '
-        f' --e s/CAPSULE_2/{setup_capsules["capsule_2"].hostname}/g '
+    module_haproxy.execute(
+        f'sed -i -e s/CAPSULE_1/{setup_capsules[0].hostname}/g '
+        f' --e s/CAPSULE_2/{setup_capsules[1].hostname}/g '
         f' /etc/haproxy/haproxy.cfg'
     )
-    haproxy.execute('systemctl restart haproxy.service')
-    haproxy.execute('mkdir /var/lib/haproxy/dev')
-    haproxy.session.sftp_write(
+    module_haproxy.execute('systemctl restart haproxy.service')
+    module_haproxy.execute('mkdir /var/lib/haproxy/dev')
+    module_haproxy.session.sftp_write(
         source=DataFile.DATA_DIR.joinpath('99-haproxy.conf'),
         destination='/etc/rsyslog.d/99-haproxy.conf',
     )
-    haproxy.execute('setenforce permissive')
-    result = haproxy.execute('systemctl restart haproxy.service rsyslog.service')
-    assert result.status == 0
-
-    return {'haproxy': haproxy}
-
-
-@pytest.fixture(scope='module')
-def loadbalancer_setup(
-    module_org,
-    content_for_client,
-    setup_capsules,
-    module_target_sat,
-    setup_haproxy,
-    module_location,
-):
-    lb_hostname = setup_haproxy['haproxy'].hostname
-    haproxy_url = f'https://{lb_hostname}:9090'
-
-    for capsule in setup_capsules.values():
-        # Enable Registration and Template Plugins
-        opts = {
-            'foreman-proxy-registration': 'true',
-            'foreman-proxy-templates': 'true',
-            'foreman-proxy-registration-url': f'{haproxy_url}',
-            'foreman-proxy-template-url': f'{haproxy_url}',
-        }
-        capsule.install(InstallerCommand(installer_opts=opts))
-
-        module_target_sat.cli.Capsule.update(
-            {
-                'name': capsule.hostname,
-                'organization-ids': module_org.id,
-                'location-ids': module_location.id,
-            }
+    module_haproxy.execute('setenforce permissive')
+    assert module_haproxy.execute('systemctl restart haproxy.service rsyslog.service').status == 0
+
+    haproxy_url = f'https://{module_haproxy.hostname}:9090'
+    # Enable Registration and Template Plugins
+    opts = {
+        'foreman-proxy-registration': 'true',
+        'foreman-proxy-templates': 'true',
+        'foreman-proxy-registration-url': f'{haproxy_url}',
+        'foreman-proxy-template-url': f'{haproxy_url}',
+    }
+    for capsule in setup_capsules:
+        assert capsule.install(InstallerCommand(installer_opts=opts)).status == 0, (
```

Comment:
Here.
`capsule` -> `setup_capsules` -> `module_lb_capsules` which is a module scoped fixture. That capsule may be used in the future by some other test (e.g. using `module_lb_capsules` fixture) and if that test runs after this installer run, and that installer run breaks something, that test may get a broken capsule (not Satellite).